waspc/src/Analyzer/Evaluator.hs

```diff
@@ -1,15 +1,50 @@
+{-# LANGUAGE LambdaCase #-}
+
 module Analyzer.Evaluator
-  ( EvaluationError,
+  ( EvaluationError (..),
     evaluate,
+    Decl,
+    takeDecls,
+    module Analyzer.Evaluator.Types,
   )
 where
 
-import Analyzer.Decl (Decl)
-import Analyzer.TypeChecker (TypedAST)
-import Analyzer.TypeDefinitions (TypeDefinitions)
+import Analyzer.Evaluator.Decl (Decl, takeDecls)
+import Analyzer.Evaluator.EvaluationError
+import Analyzer.Evaluator.Types
+import Analyzer.Type
+import qualified Analyzer.TypeChecker.AST as AST
+import qualified Analyzer.TypeDefinitions as TD
+import Control.Monad.Except
+import Control.Monad.Reader
+import Control.Monad.State
+import qualified Data.HashMap.Strict as H
+import Data.Maybe (fromMaybe)
 
-data EvaluationError
+type Eval a = StateT (H.HashMap String Decl) (ReaderT TD.TypeDefinitions (Except EvaluationError)) a
```

Comment:
I would move this below `evaluate`, since `evaluate` is top level function and doesn't need `Eval`.

Reply:
Also, let's alias this HashMap with `type` so it is clearer what it is.

Reply:
Done